test/python/tools/test_compiler.py

```diff
@@ -342,6 +341,47 @@ def test_compile_pass_manager(self):
         rfalse = backend.run(qrfalse).result()
         self.assertEqual(rtrue.get_counts(), rfalse.get_counts())
 
+    def test_compile_with_initial_layout(self):
+        """Test compile with an initial layout.
+        Regression test for #1711
+        """
+        qr = QuantumRegister(3)
+        cr = ClassicalRegister(3)
+        qc = QuantumCircuit(qr, cr)
+        qc.u1(3.14, qr[0])
+        qc.cx(qr[1], qr[0])
+        qc.barrier(qr)
+        qc.measure(qr, cr)
+        backend = FakeRueschlikon()
+        initial_layout = {0: (qr, 1), 2: (qr, 0), 15: (qr, 2)}
+
+        qobj = compile(qc, backend, seed=42, initial_layout=initial_layout)
+
+        compiled_ops = qobj.experiments[0].instructions
+        for operation in compiled_ops:
+            if operation.name == 'cx':
+                self.assertIn(operation.qubits, backend.configuration().coupling_map)
+                self.assertIn(operation.qubits, [[15, 0], [15, 2]])
+
+    def test_execute_with_initial_layout(self):
+        """ Test execute with initial layout
+        Regression test for #1637
+        """
+        qr = QuantumRegister(3)
+        cr = ClassicalRegister(3)
+        qc = QuantumCircuit(qr, cr)
+        qc.u1(3.14, qr[0])
+        qc.cx(qr[1], qr[0])
+        qc.barrier(qr)
+        qc.measure(qr, cr)
+        backend = BasicAer.get_backend('qasm_simulator')
```

Comment:
initial_layout doesn't really make sense for a simulator backend, so can we also modify this test to use a FakeDevice backend? also this test isn't asserting anything. I'm fine just making this test a variant of the above compile test. Execute is a thin wrapper around compile anyway.